x-pack/plugin/ilm/src/internalClusterTest/java/org/elasticsearch/xpack/ilm/ILMHistoryTests.java

```diff
@@ -0,0 +1,92 @@
+        Settings.Builder settings = Settings.builder().put(super.nodeSettings(nodeOrdinal));
+        settings.put(XPackSettings.MACHINE_LEARNING_ENABLED.getKey(), false);
+        settings.put(XPackSettings.SECURITY_ENABLED.getKey(), false);
+        settings.put(XPackSettings.WATCHER_ENABLED.getKey(), false);
+        settings.put(XPackSettings.GRAPH_ENABLED.getKey(), false);
+        settings.put(LifecycleSettings.LIFECYCLE_POLL_INTERVAL, "1s");
+        settings.put(LifecycleSettings.SLM_HISTORY_INDEX_ENABLED_SETTING.getKey(), false);
+        return settings.build();
+    }
+
+    @Override
+    protected boolean ignoreExternalCluster() {
+        return true;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(LocalStateCompositeXPackPlugin.class, IndexLifecycle.class);
+    }
+
+    private void putTestPolicy() throws InterruptedException, java.util.concurrent.ExecutionException {
+        LifecyclePolicy lifecyclePolicy = randomTimeseriesLifecyclePolicy("test");
+        PutLifecycleAction.Request putLifecycleRequest = new PutLifecycleAction.Request(lifecyclePolicy);
+        PutLifecycleAction.Response putLifecycleResponse = client().execute(PutLifecycleAction.INSTANCE, putLifecycleRequest).get();
+        assertAcked(putLifecycleResponse);
+    }
+
+    public void testIlmHistoryIndexCanRollover() throws Exception {
+        putTestPolicy();
+        Settings settings = Settings.builder().put(indexSettings()).put(SETTING_NUMBER_OF_SHARDS, 1)
+            .put(SETTING_NUMBER_OF_REPLICAS, 0).put(LifecycleSettings.LIFECYCLE_NAME, "test").build();
+        CreateIndexResponse res = client().admin().indices().prepareCreate("test").setSettings(settings).get();
+        assertTrue(res.isAcknowledged());
+
+        assertBusy(() -> {
+            String firstIndex = ILMHistoryStore.ILM_HISTORY_INDEX_PREFIX + "000001";
+            try {
+                GetIndexResponse getIndexResponse = client().admin().indices().prepareGetIndex().setIndices(firstIndex).get();
+                assertThat(getIndexResponse.getIndices(), arrayContaining(firstIndex));
+            } catch (Exception e) {
+                fail(e.getMessage());
+            }
+        });
+
+        RolloverResponse rolloverResponse = client().admin().indices().prepareRolloverIndex(ILMHistoryStore.ILM_HISTORY_ALIAS).get();
+        String secondIndex = ILMHistoryStore.ILM_HISTORY_INDEX_PREFIX + "000002";
+        assertTrue(rolloverResponse.isAcknowledged());
+        assertThat(rolloverResponse.getNewIndex(), is(secondIndex));
+
+        GetIndexResponse getIndexResponse = client().admin().indices().prepareGetIndex().setIndices(secondIndex).get();
+        assertThat(getIndexResponse.getIndices(), arrayContaining(secondIndex));
+
+        //wait for all history items to index to avoid waiting for timeout in ILMHistoryStore beforeBulk
+        Thread.sleep(6000);
```

Comment:
why would `beforeBulk` timeout? would removing the policy from the `test` index and/or stopping ILM solve the issue of outstanding ilm-history entries? (we're manually rolling over the `ilm-history` alias so ILM is only needed to trigger the creation of the first history index)

Reply:
I've changed it, I stop ILM to reduce amount of history items to index and I wait for the ones that are created. It should be better than explicit wait. WDYT?